### PR TITLE
Fix comparing Istio version for download

### PR DESCRIPTION
**Pull request by shamsher31**
To fix https://github.com/istio/istio/issues/32194

### CODE REVIEW

File: release/downloadIstioCandidate.sh [hunk 1]
```diff
@@ -109,7 +109,7 @@ ARCH_UNSUPPORTED="1.5"
 
 if [ "${OS}" = "Linux" ] ; then
   # This checks if 1.6 <= 1.5 or 1.4 <= 1.5
-  if [ "$(expr "${ARCH_SUPPORTED}" \<= "${ARCH_UNSUPPORTED}")" -eq 1 ]; then
+  if awk 'BEGIN {exit !('$ARCH_SUPPORTED' <= '$ARCH_UNSUPPORTED')}'; then
```
Comment (shamsher31) [inline release/downloadIstioCandidate.sh:112]: @ericvn After a couple of tests, I'm afraid this is still not comparing float value properly. It only compares floating-point numbers if both have the same decimal points. so `1.5 >= 1.6` pass but `1.5 >= 1.10` fails. Any better idea?

Comment (ericvn) [reply release/downloadIstioCandidate.sh:112]: A version compare is more than a few lines if in bash only. I wonder if we could assume there was a `sort` command that supported version sort. Makes it easier, but might not work everywhere.

Comment (ericvn) [reply release/downloadIstioCandidate.sh:112]: It seems a number of tools who used to download a script and run it have mostly changed to just having one download the actual binary. I believe `kubectl` used to run a script in the past and has moved to just getting the actual download.

Comment (shamsher31) [reply release/downloadIstioCandidate.sh:112]: > A version compare is more than a few lines if in bash only. I wonder if we could assume there was a sort command that supported version sort. Makes it easier, but might not work everywhere.

I have tried few options that work in `bash` but not in `shell` (bash seems easier). 

Wondering if we still support download lower than 1.5 (without arch support)? that is the problem.

>  I believe kubectl used to run a script in the past and has moved to just getting the actual download.

Maybe we can also have multiple conditions for different download options. For 1.5 and lower which will be used rarely as compared to versions greater than 1.6 and above.

Comment (shamsher31) [reply release/downloadIstioCandidate.sh]: @ericvn I applied a hack for floating-point comparison with 3 digits and now it's working as expected. Let me know if this looks good.

**Comment by ericvn** [comment]
@shamsher31 I think we can use `sort --version-sort` since it's already used in the script.
So, something like:
```
if [ $(printf '%s\n%s' "${ARCH_UNSUPPORTED}" "${ARCH_SUPPORTED}" | sort --version-sort | head -n 1) = "${ARCH_UNSUPPORTED}" ]; then
  echo Is greater or equal
else
  echo Is less than
fi
```
I haven't tried to verify the edge condition.

**Comment by ericvn** [comment]
A test.sh:
```
#!/bin/sh

# Istio 1.5 and below do not have arch support
ARCH_UNSUPPORTED="1.5"
if [ $(printf '%s\n%s' "${ARCH_UNSUPPORTED}" "${ISTIO_VERSION}" | sort --version-sort | head -n 1) = "${ARCH_UNSUPPORTED}" ]; then
  echo ISTIO_VERSION \("${ISTIO_VERSION}"\) is equal or greater than "${ARCH_UNSUPPORTED}"
else
  echo ISTIO_VERSION \("${ISTIO_VERSION}"\) is less than "${ARCH_UNSUPPORTED}"
fi
```

**Comment by ericvn** [comment]
Just looking at the comments, might need to change `1.5` to `1.6` for the edge condition using my snippet.

**Comment by ericvn** [comment]
I did check my changes in https://github.com/istio/istio/pull/32371

**Comment by shamsher31** [comment]
> I did check my changes in #32371
Sorry, I was on PTO yesterday so didn't get a chance to look into it. Thanks for your PR, I will close this one.
